src/Umbraco.Core/StringExtensions.cs

```diff
@@ -1465,10 +1470,7 @@ internal static string ToValidXmlString(this string text)
         /// <returns></returns>
         internal static Guid ToGuid(this string text)
         {
-            var md5 = MD5.Create();
-            byte[] myStringBytes = Encoding.ASCII.GetBytes(text);
-            byte[] hash = md5.ComputeHash(myStringBytes);
-            return new Guid(hash);
+            return new Guid(text.GenerateHash());

```

Comment:
OK, I added this method and forced SHA-1 and the unit tests passed and everything looks like a valid GUID.  In fact, it may be more valid because it actually adheres to the actually RFC 4122 spec.

I'm pushing this to Github now and will update this pull request.